src/internal/signalr/handler.go

```diff
@@ -344,7 +344,8 @@ func (h *observationsHandler) handleOutPhase(observation model.Observation) erro
 
 	outPhaseType := model.OutputPhaseType(val).ToFimpState()
 
-	if outPhaseType == "" && !h.cache.IsSessionFinished() {
+	// Charger sets outPhaseType parameter to "" if charger not charging, even if it has ongoing charging session.
+	if outPhaseType == "" && (h.cache.HasOngoingChargingSession() || h.cache.TotalPower() > 0) {
```

Comment:
Hmm, maybe we should skip this total power check actually? I'd test that with a physical charger to see if there's an actual difference.